resolver/resolver.go

```diff
@@ -51,8 +47,34 @@ func New(version string, config records.Config) *Resolver {
 		rs:      &records.RecordGenerator{},
 		rng:     rand.New(rand.NewSource(time.Now().UnixNano())),
 	}
-	r.extResolver = r.defaultExtResolver
+
 	r.startZKdetection = startDefaultZKdetector
+
+	if !config.ExternalOn {
+		return r
+	}
+
+	timeout := 5 * time.Second
+	if config.Timeout != 0 {
+		timeout = time.Duration(config.Timeout) * time.Second
+	}
+
+	clients := make([]exchanger.Exchanger, 2)
+	for i, proto := range [...]string{"udp", "tcp"} {
+		clients[i] = exchanger.Decorate(&dns.Client{
+			Net:          proto,
+			DialTimeout:  timeout,
+			ReadTimeout:  timeout,
+			WriteTimeout: timeout,
+		}, exchanger.ErrorLogging(logging.Error))
+	}
+
+	r.extResolver = exchanger.Decorate(
+		exchanger.Multi(clients...),
```

Comment:
This approach to support https://tools.ietf.org/html/rfc5966 isn't the best. We should instead check the `Truncated` bit and re-try with TCP. Current status, reading: https://tools.ietf.org/html/rfc5966 and https://tools.ietf.org/html/rfc6891